### PR TITLE
Add more tests

### PR DESCRIPTION
**Pull request by kpp**
The tests should be fixed to match you u8::MAX/u16::MAX (and possibly u32::MAX) issues.

### CODE REVIEW

File: tests/nano_arena.rs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+use compact_arena::in_nano_arena;
+
+#[test]
+fn add_255_objects() {
+    in_nano_arena!(arena, {
+        for _ in 0..255 {
```
Comment (llogiq) [inline tests/nano_arena.rs]: I changed NanoArena to allow for 256 elements. Also we can now use `mk_nano_arena(arena);` without requiring a new scope.

File: tests/nano_arena.rs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        }
+    });
+}
+
+#[cfg(not(miri))] // miri will fail due to failed assertion idx < NANO_ARENA_ITEMS
+#[test]
+#[should_panic(expected = "assertion failed: (i as usize) < NANO_ARENA_ITEMS")]
+fn add_256_objects() {
+    in_nano_arena!(arena, {
+        for _ in 0..255 {
```
Comment (llogiq) [inline tests/nano_arena.rs]: same here.

File: tests/tiny_arena.rs [hunk 3]
```diff
@@ -0,0 +1,26 @@
+
+#[cfg(not(miri))] // this will take too much time for miri
+#[test]
+fn add_65535_objects() {
+    use compact_arena::in_tiny_arena;
+
+    in_tiny_arena!(arena, {
+        for _ in 0..65535 {
```
Comment (llogiq) [inline tests/tiny_arena.rs]: Like with `NanoArena`, `TinyArena' now can hold 65536 elements and can be created with `mk_tiny_arena!(arena);`. Please update the test accordingly.

File: tests/tiny_arena.rs [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+}
+
+#[cfg(not(miri))] // miri will fail due to failed assertion idx < TINY_ARENA_ITEMS
+#[test]
+#[should_panic(expected = "assertion failed: (i as usize) < TINY_ARENA_ITEMS")]
+fn add_65536_objects() {
+    use compact_arena::in_tiny_arena;
+
+    in_tiny_arena!(arena, {
+        for _ in 0..65535 {
```
Comment (llogiq) [inline tests/tiny_arena.rs]: Same here.

**Comment by kpp** [comment]
Also see https://doc.rust-lang.org/1.7.0/book/macros.html#the-variable-crate

**Comment by llogiq** [comment]
Thank you!

**Pull request merged**